dev/com.ibm.ws.cdi.weld/src/com/ibm/ws/cdi/impl/weld/WebSphereCDIDeploymentImpl.java

```diff
@@ -494,10 +494,10 @@ public Iterable<Metadata<Extension>> getExtensions() {
                 }
             }
             //if the probe is enabled, add the probe extension
-
-            if (CDIUtils.isDevelopementMode()) {
-                extensionSet.add(getProbeExtension());
-                WebSphereBeanDeploymentArchive bda = getBeanDeploymentArchive(ProbeExtension.class);
+            WeldDevelopmentMode devMode = this.cdiRuntime.getWeldDevelopmentMode();
+            if (devMode != null) {
+                extensionSet.add(devMode.getProbeExtension());
+                WebSphereBeanDeploymentArchive bda = devMode.getProbeBDA(this);
                 extensionBDAs.put(bda.getId(), bda);
```

Comment:
Why do we need to add the probe extension BDA in two places? (Both here and in CDIContainerImpl)

Reply:
The two places are performing different functions. The CDIContainerImpl code is creating the Probe BDA. This code is getting the Extension Metadata.